programs/create-genesis/genesis_create.cpp

```diff
@@ -1003,6 +1018,120 @@ struct genesis_create::genesis_create_impl final {
     }
 
+    name get_name(const golos::account_name_type& acc) {
+        return generate_name(acc.value(_accs_map));
+    }
+
+    void store_posts() {
+        std::cout << "Creating reward pool, posts & votes..." << std::endl;
+
+        // store pool
+        const auto& gp = _visitor.gpo;
+        const int n = _visitor.comments.size();     // messages count
+        primary_key_t pk = 0;                       // created
+        db.start_section(gls_post_account_name, N(rewardpools), "rewardpool", 1);
+        db.insert(pk, gls_post_account_name, mvo
+            ("created", pk)
+            ("rules", _info.params.posting_rules)
+            ("state", mvo
+                ("msgs", n)
+                ("funds", gp.total_reward_fund_steem)
+                ("rshares", to_fwide(gp.total_reward_shares2))
+                ("rsharesfn", to_fwide(gp.total_reward_shares2))
+            )
+        );
+
+        // first create lookup table to find author by post id
+        fc::flat_map<uint64_t, name> authors;           // post_id:name
+        for (const auto& c : _visitor.comments) {
+            authors[c.second.id] = get_name(c.second.author);
+        }
+
+        // store votes
+        fc::flat_map<uint64_t, uint64_t> vote_weights_sum;
+        db.start_section(gls_post_account_name, N(vote), "voteinfo", _visitor.votes.size());
+        for (const auto& v: _visitor.votes) {
+            std::vector<mvo> delegators;
+            for (const auto& d: v.delegator_vote_interest_rates) {
+                delegators.emplace_back(mvo
+                    ("delegator", get_name(d.account))
+                    ("interest_rate", d.interest_rate)
+                    ("payout_strategy", int(d.payout_strategy))
+                );
+            }
+            auto cid = v.comment;
+            uint64_t w = v.orig_rshares <= 0 ? 0 : v.rshares;
+            vote_weights_sum[cid] += w;
+            if (v.num_changes != 0) {
+                w = 0;
+            } else if (w > 0) {
+                uint64_t auction_window = _visitor.comments[cid].active.auction_window_size;
+                if (v.auction_time != auction_window) {
+                    w = int_arithmetic::safe_prop(w, static_cast<uint64_t>(v.auction_time), auction_window);
+                }
+            }
+            db.insert(pk, authors[cid], mvo
+                ("id", pk)            // uint64_t
+                ("message_id", cid)    // uint64_t
+                ("voter", generate_name(_accs_map[_visitor.acc_id2idx[v.voter]]))         // name
+                ("weight", v.vote_percent)        // int16_t
+                ("time", time_point(v.last_update).time_since_epoch().count())          // uint64_t
+                ("count", v.num_changes)         // uint8_t
+                ("delegators", delegators)    // std::vector<delegate_voter>
+                ("curatorsw", w)     // base_t
+                ("rshares", to_fbase(v.rshares))       // base_t
```

Comment:
after [this](https://github.com/GolosChain/golos.contracts/pull/610/commits/551b85a6af2029bd4a64f8831465585ca9885ab9) commit, rshares is interpreted as a raw value

Reply:
thx, fixed in 28683ce